tests/ops/api/v1/test_main.py

```diff
@@ -1,13 +1,120 @@
-    """Mock Non-HTTP Exception"""
-
-    pass
+from fides.api.main import configure_security_env_overrides, create_fides_app
+from fides.api.ops.util.oauth_util import (
+    get_root_client,
+    verify_oauth_client,
+    verify_oauth_client_cli,
+)
 
 
 def test_read_autogenerated_docs(api_client: TestClient):
     """Test to ensure automatically generated docs build properly"""
     response = api_client.get(f"/openapi.json")
     assert response.status_code == 200
+
+
+class TestConfigureSecurityEnvOverrides:
+    def test_configure_security_env_overrides_dev(self) -> None:
+        """
+        This test verifies that when set to 'dev', only the
+        cli-related endpoints default to the root user.
+        """
+        security_env = "dev"
+        test_app = FastAPI(title="test")
+        test_app = configure_security_env_overrides(
+            application=test_app, security_env=security_env
+        )
+        assert test_app.dependency_overrides[verify_oauth_client_cli] == get_root_client
+
+    def test_configure_security_env_overrides_demo(self) -> None:
+        """
+        This test verifies that when set to 'demo', both oauth client
+        checks get overwritten.
+        """
+        security_env = "demo"
+        test_app = FastAPI(title="test")
+        test_app = configure_security_env_overrides(
+            application=test_app, security_env=security_env
+        )
+        assert test_app.dependency_overrides[verify_oauth_client] == get_root_client
+        assert test_app.dependency_overrides[verify_oauth_client_cli] == get_root_client
+
+    def test_configure_security_env_overrides_prod(self) -> None:
+        """
+        This test verifies that when set to 'prod', there are no
+        dependency overrides for the oauth clients.
+        """
+        security_env = "prod"
+        test_app = FastAPI(title="test")
+        test_app = configure_security_env_overrides(
+            application=test_app, security_env=security_env
+        )
+        with pytest.raises(KeyError):
+            test_app.dependency_overrides[verify_oauth_client]
+
+        with pytest.raises(KeyError):
+            test_app.dependency_overrides[verify_oauth_client_cli]
+
+
+class TestSecurityEnvs:
+    @pytest.mark.parametrize("endpoint", ["user", "system"])
+    @pytest.mark.asyncio
+    def test_demo_endpoints_unprotected(self, endpoint: str) -> None:
+        """
+        Test that all endpoints are unprotected in a 'demo'-tier
+        security environment.
+        """
+        security_env = "demo"
+        test_app = create_fides_app()
+        test_app = configure_security_env_overrides(
+            application=test_app, security_env=security_env
+        )
+        with TestClient(test_app) as test_client:
+            response = test_client.get(f"/api/v1/{endpoint}")
+            print(response.url)
+            print(response.text)
+            assert response.ok
+
+    def test_dev_endpoints_protected(self) -> None:
```

Comment:
@sanders41 @allisonking any guidance here on what might be going wrong? I'm using the test client, and the first test _always_ passes, but the rest fail and complain about the event loop. Any idea how to potentially solve this? I've googled around but haven't found anything that's worked

Reply:
The `event_loop` fixture should be putting everything in the same loop and preventing this. https://github.com/ethyca/fides/blob/502d689676d5bd77b14875e1e1c3e80b5feb2143/tests/ops/conftest.py#L131

If you remove `@pytest.mark.asyncio` do you still get the error? That decorator lets the test itself be async so `async def test_demo_endpoints_unpretected...```. I wonder if the error is that the test doesn't get awaited since it isn't async.

Reply:
yeah I do, I added that as an attempt to fix things...but its not necessary when using `TestClient` technically since it runs things synchronously

Reply:
Are you able to successfully run this endpoint outside of tests? I tried isolating one of the tests with `pytest -k  test_dev_endpoints_protected`and not using fixtures but still got the same error. The error always says postgres has the event loop issue.

I think the answer here is going to be yes, I just want to make sure so we know which path to go down.

```py
from httpx import AsyncClient

...

    @pytest.mark.asyncio
    async def test_dev_endpoints_protected(self) -> None:
        """
        Test that the non-cli-related endpoints are protected in
        a 'dev'-tier security environment.
        """
        security_env = "dev"
        endpoint = "data_category"
        test_app = create_fides_app()
        test_app = configure_security_env_overrides(
            application=test_app, security_env=security_env
        )
        async with AsyncClient(
            app=test_app, base_url="http://0.0.0.0:8080", follow_redirects=True
        ) as client:
            response = await client.get(f"/api/v1/{endpoint}")
            print(response.url)
            print(response.text)
            assert response.status_code == 200
```

Reply:
@sanders41 I'm going to remove these tests from this PR and add them to #2463 since we can manually verify these changes here, and then those tests will help prevent regressions in the future

Reply:
yep, I've confirmed the endpoints work via manual testing